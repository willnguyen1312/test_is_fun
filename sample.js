// @ts-nocheck
/* eslint-disable */

// import R from 'ramda';

const R = require('ramda');



export const getEmail = R.map(R.prop("email"));

export const filterAdmin = R.filter(R.propEq("role", "admin"));

export const getAdminEmails = R.pipe(filterAdmin, getEmail);

// const M = require('ramda-fantasy').Maybe;
// const linearScale = (domainMin,domainMax) =>
//                       (rangeMin,rangeMax) =>
//                         (n) => 
//                           rangeMin + (n - domainMin) * (rangeMax - rangeMin) / (domainMax - domainMin)

// const initialScaleData = [0, 1000, 3000, 2000, 5000, 4000, 7000, 6000, 9000, 8000, 10000];
// // let linearScale = linearScale([0,1000]);
// let converter = linearScale(0, 1000)(0,100);

// console.log(R.map(converter(2),initialScaleData))

// const Just    = M.Just;
// const Nothing = M.Nothing;

// const safeDiv = R.curry((n, d) => d === 0 ? Nothing() : Just(n / d));
// const lookup = R.curry((k, obj) => k in obj ? Just(obj[k]) : Nothing());

// // getOrElse :: Maybe a ~> a -> a -- provide a default value if Nothing
// lookup('foo', { foo: 'bar' }).getOrElse('baz'); // 'bar'
// lookup('foo', { abc: 'bar' }).getOrElse('baz'); // 'baz'

// // maybe :: b -> (a -> b) -> Maybe a -> b -- transform the value if it exists
// //                                        -- with the provided function,
// //                                        -- otherwise return the default value
// plus1or0 = M.maybe(0, R.inc);
// console.log(plus1or0(safeDiv(42, 2)));; // 22
// console.log(plus1or0(safeDiv(42, 0)));; // 0


//=> 'ghijklm'
// var fn = R.cond([
//   [R.view(R.lensProp('c')),   R.always('3')],
//   [R.equals(100), R.always('water boils at 100°C')],
//   [R.T,           temp => 'nothing special happens at ' + temp + '°C']
// ]);
// console.log(fn({
//   a: 1,
//   b: null
// }));



// var abby = {name: 'Abby', age: 7, hair: 'blond'};
// var fred = {name: 'Fred', age: 12, hair: 'brown'};
// var rusty = {name: 'Rusty', age: 10, hair: 'brown'};
// var alois = {name: 'Alois', age: 15, disposition: 'surly'};
// var kids = [abby, fred, rusty, alois];
// var hasBrownHair = R.propEq('hair', 'blond');
// console.log(R.filter(hasBrownHair, kids)); //=> [fred, rusty]





// total =[1,2,3,4] => [2,3,4]

// display(total) [1..4] => [2..4]

// head(total) => 1