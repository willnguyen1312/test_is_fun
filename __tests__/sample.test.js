/* eslint-disable */
/*
  eslint fp/no-unused-expression: 0,
  fp/no-nil: 0,
  better/explicit-return: 0
*/
// import { data } from '../data';
// Happy input
const data = [
  {
    role: "admin",
    email: "admin1",
  },
  {
    role: "admin",
    email: "admin2",
  },
  {
    role: "normal",
    email: "normal1",
  },
  {
    role: "normal",
    email: "normal2",
  },
];
import { getAdminEmails, getEmail, filterAdmin } from "../sample";

// @ts-ignore
// expect.addSnapshotSerializer({
//   test: val => val.email && val.role,
//   print: val => `"Role: ${val.role} - Email: ${val.email}"`,
// });

describe("Sample", () => {
  it("Should return email of admin", () => {
    expect(getAdminEmails(data)).toMatchSnapshot();
  });

  it("Should return email", () => {
    expect(getEmail(data)).toMatchSnapshot();
  });

  it("Should filter admin", () => {
    expect(filterAdmin(data)).toMatchSnapshot();
  });
});
