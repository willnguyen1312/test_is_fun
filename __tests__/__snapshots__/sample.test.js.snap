// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample Should filter admin 1`] = `
Array [
  Object {
    "email": "admin1",
    "role": "admin",
  },
  Object {
    "email": "admin2",
    "role": "admin",
  },
]
`;

exports[`Sample Should return email 1`] = `
Array [
  "admin1",
  "admin2",
  "normal1",
  "normal2",
]
`;

exports[`Sample Should return email of admin 1`] = `
Array [
  "admin1",
  "admin2",
]
`;
